pkg/webhook/cluster/nutanix_validator_test.go

```diff
@@ -221,3 +226,196 @@ func TestValidatePrismCentralIPDoesNotEqualControlPlaneIP(t *testing.T) {
+			cluster:       fakeCluster(t, true, false, false),
+			expectedErr:   true,
+			expectedErrMessages: []string{
+				"spec.topology.variables.clusterConfig.value.controlPlane.nutanix.machineDetails.cluster: Required value: \"cluster\" must set when failureDomains is not configured.", //nolint:lll // Message is long.
+				"spec.topology.variables.clusterConfig.value.controlPlane.nutanix.machineDetails.subnets: Required value: \"subnets\" must set when failureDomains is not configured.", //nolint:lll // Message is long.
+			},
+		},
+		{
+			name:                "controlPlane failureDomains configured and missing machineDetail cluster and subnets",
+			clusterConfig:       fakeClusterConfigSpec(true, false, false),
+			cluster:             fakeCluster(t, true, false, false),
+			expectedErr:         false,
+			expectedErrMessages: nil,
+		},
+		{
+			name:                "controlPlane failureDomains configured and machineDetail cluster and subnets set",
+			clusterConfig:       fakeClusterConfigSpec(true, true, true),
+			cluster:             fakeCluster(t, true, false, false),
+			expectedErr:         false,
+			expectedErrMessages: nil,
+		},
+		{
+			name:          "worker failureDomain not configured and missing machineDetail cluster",
+			clusterConfig: fakeClusterConfigSpec(true, false, false),
+			cluster:       fakeCluster(t, false, false, true),
+			expectedErr:   true,
+			expectedErrMessages: []string{
+				"spec.topology.workers.machineDeployments.variables.overrides.workerConfig.value.nutanix.machineDetails.cluster: Required value: \"cluster\" must set when failureDomain is not configured.", //nolint:lll // Message is long.
+			},
+		},
+		{
+			name:          "worker failureDomain not configured and missing machineDetail subnets",
+			clusterConfig: fakeClusterConfigSpec(true, false, false),
+			cluster:       fakeCluster(t, false, true, false),
+			expectedErr:   true,
+			expectedErrMessages: []string{
+				"spec.topology.workers.machineDeployments.variables.overrides.workerConfig.value.nutanix.machineDetails.subnets: Required value: \"subnets\" must set when failureDomain is not configured.", //nolint:lll // Message is long.
+			},
+		},
+		{
+			name:          "worker failureDomain not configured and missing machineDetail cluster and subnets",
+			clusterConfig: fakeClusterConfigSpec(true, false, false),
+			cluster:       fakeCluster(t, false, false, false),
+			expectedErr:   true,
+			expectedErrMessages: []string{
+				"spec.topology.workers.machineDeployments.variables.overrides.workerConfig.value.nutanix.machineDetails.cluster: Required value: \"cluster\" must set when failureDomain is not configured.", //nolint:lll // Message is long.
+				"spec.topology.workers.machineDeployments.variables.overrides.workerConfig.value.nutanix.machineDetails.subnets: Required value: \"subnets\" must set when failureDomain is not configured.", //nolint:lll // Message is long.
+			},
+		},
```

Comment:
There are few cases to this:
* top-level workerConfig does not exist and only variable overrides exist
* top-level workerConfig exists but variable overrides don't exist
* top-level workerConfig and variable overrides both exist